### ISSUE TITLE
correct curl patch url is not generated by python requests

### ISSUE THREAD

**Issue by zerolaser** [NickCarneiro/curlconverter#68]
this is the sample code. which i  was trying to convert to python, but is showing up POST instead of PATCH
```
curl 'https://ci.example.com/go/api/agents/adb9540a-b954-4571-9d9b-2f330739d4da' \
      -u 'username:password' \
      -H 'Accept: application/vnd.go.cd.v4+json' \
      -H 'Content-Type: application/json' \
      -X PATCH \
      -d '{
        "hostname": "agent02.example.com",
        "agent_config_state": "Enabled",
        "resources": ["Java","Linux"],
        "environments": ["Dev"]
        }'
 ```

**Comment by zerolaser** [comment]
@NickCarneiro  Also add the -v verbose option. whenever we use this option this should output the debug http information.

**Comment by NickCarneiro** [comment]
fixed in 1.13.0.

Support for `-v` is an interesting idea. Right now we don't even print out responses.

**Comment by zerolaser** [comment]
@NickCarneiro thanks for the fix.
